core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -359,4 +539,49 @@ class SpendReportingService(
         case ex: Exception =>
           Future.failed(RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, ex)))
       }
+
+  def getSpendForAllWorkspaces(
+    start: DateTime,
+    end: DateTime
+  ): Future[SpendReportingResults] = {
+    validateReportParameters(start, end)
+    for {
+      billingMap <- getBillingWithSpendPermission()
+      projectNames: Map[GoogleProjectId, WorkspaceName] = billingMap.values.flatten.toMap
+
+      query = getAllUserWorkspaceQuery(billingMap)
+      queryJob = setUpAllUserWorkspaceQuery(query, start, end)
+
+      job: Job <- bigQueryService.use(_.runJob(queryJob)).unsafeToFuture().map(_.waitFor())
+      _ = logSpendQueryStats(job.getStatistics[JobStatistics.QueryStatistics])
+      result = job.getQueryResults()
+    } yield result.getValues.asScala.toList match {
+      case Nil =>
+        throw RawlsExceptionWithErrorReport(
+          StatusCodes.NotFound,
+          s"no spend data found between dates ${toISODateString(start)} and ${toISODateString(end)}"
+        ) // TODO update this
+      case rows => extractCrossBillingProjectSpendReportingResults(rows, start, end, projectNames)
+    }
+  }
+
+  def getBillingWithSpendPermission(
+  ): Future[Map[BillingProjectSpendExport, Seq[(GoogleProjectId, WorkspaceName)]]] =
+    for {
+      billingProjectResources <- samDAO.listResourcesWithActions(SamResourceTypeNames.billingProject,
+                                                                 SamBillingProjectActions.readSpendReport,
+                                                                 ctx
+      )
+      billingProjectIds = billingProjectResources.map(resource => RawlsBillingProjectName(resource.resourceId)).toList
+      groupedWorkspaces <- workspaceServiceConstructor(ctx).getWorkspacesByBillingProjects(billingProjectIds)
+      gcpOnlyGroupedWorkspaces = groupedWorkspaces
+        .map { case (key, workspaces) =>
+          key -> workspaces.filter(_.workspaceType == WorkspaceType.RawlsWorkspace)
+        }
+        .filter { case (_, workspaces) => workspaces.nonEmpty }
+      // Only use the BPs we know exist in the DB and are GCP
+      spendConfigs <- getSpendExportConfigurations(gcpOnlyGroupedWorkspaces.keys.toList)
+    } yield spendConfigs.map { config =>
+      config -> gcpOnlyGroupedWorkspaces(config.billingProjectName).map(ws => (ws.googleProjectId, ws.toWorkspaceName))
+    }.toMap
```

Comment:
This implementation makes sense to me, and matches what we talked about. I don't know, though, if it aligns with the letter of the jira ticket "all workspaces where the user has owner access permissions or higher". Is that an important Product requirement? Or does this implementation make even more sense than the requirement?

Also … this should be paginated, according to the jira ticket

Reply:
IIRC it was an important product requirement to expand the access to spend reports to workspace owners, but I'll let @admull confirm that. The `readSpendReport` action is only granted to billing project owners, so this implementation definitely does make sense and is more in line with the access controls on the existing spend report API, but I think we wanted more folks to be able to see how much money they were spending

Reply:
Yes, the purpose of this feature is to give someone access to be able to see spend for any workspace where they are an Owner or above.

Reply:
> see spend for any workspace where they are an Owner or above

The current code grants access for all billing projects where the caller has the readSpendReport action. This seems appropriate and we should make sure that correct roles have this action. But this is not quite @admull's comment which talks about _workspace_ owners.

Reply:
Ok I've updated it to compare workspace ownership to `readSpendReport` permission on the billing project - by making a call to sam for BPs and another one for workspaces and intersecting the results.  Not sure if this is what we want or if it's efficient, so please let me know!

Reply:
Sorry, missed your earlier comment! Currently, users can only have the `readSpendReport` action on billing projects that they own. Since we want to let anyone with Owner access or above on a workspace see the spendReport, we should either add a new `readSpendReport` action to the `owner` role on the Sam `workspace` resource type or check the `own` action on the Sam resource